tests/unit/transformers/test_base.py

```diff
@@ -1,10 +1,374 @@
+
+        # Assert
+        expected = True
+        assert output == expected
+
+    def test_is_reverse_deterministic(self):
+        """Test the ``is_reverse_deterministic`` method.
+
+        Setup:
+            - set the ``DETERMINISTIC_REVERSE`` of the ``BaseTransformer`` to True.
+
+        Output:
+            - the booloan value stored in ``DETERMINISTIC_REVERSE``.
+        """
+        # Setup
+        base_transformer = BaseTransformer()
+        base_transformer.DETERMINISTIC_REVERSE = True
+
+        # Run
+        output = base_transformer.is_reverse_deterministic()
+
+        # Assert
+        expected = True
+        assert output == expected
+
+    def test_is_composition_identity(self):
+        """Test the ``is_composition_identity`` method.
+
+        Setup:
+            - set the ``COMPOSITION_IS_IDENTITY`` of the ``BaseTransformer`` to True.
+
+        Output:
+            - the booloan value stored in ``COMPOSITION_IS_IDENTITY``.
+        """
+        # Setup
+        base_transformer = BaseTransformer()
+        base_transformer.COMPOSITION_IS_IDENTITY = True
+
+        # Run
+        output = base_transformer.is_composition_identity()
+
+        # Assert
+        expected = True
+        assert output == expected
+
+    def test_get_next_transformers(self):
+        """Test the ``get_next_transformers`` method.
+
+        Setup:
+            - set the ``column_prefix`` of the ``BaseTransformer`` to a string.
+            - set the ``NEXT_TRANSFORMERS`` of the ``BaseTransformer`` to a dictionary.
+
+        Output:
+            - the dictionary set in ``NEXT_TRANSFORMERS`` with the ``column_prefix`` string
+            added to the beginning of the keys.
+        """
+        # Setup
+        base_transformer = BaseTransformer()
+        base_transformer.column_prefix = 'column_name'
+        base_transformer.NEXT_TRANSFORMERS = {
+            'value': 'NullTransformer'
+        }
+
+        # Run
+        output = base_transformer.get_next_transformers()
+
+        # Assert
+        expected = {
+            'column_name.value': 'NullTransformer'
+        }
+        assert output == expected
+
+    def test__store_columns_list(self):
+        """Test the ``_store_columns`` method when passed a list.
+
+        The method should store the passed columns in the ``columns`` attribute.
+
+        Input:
+            - a data frame.
+            - a list of a subset of the columns of the dataframe.
+        """
+        # Setup
+        data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4, 5, 6],
+            'c': [7, 8, 9]
+        })
+        columns = ['a', 'b']
+        base_transformer = BaseTransformer()
+
+        # Run
+        base_transformer._store_columns(columns, data)
+        stored_columns = base_transformer.columns
+
+        # Assert
+        expected = ['a', 'b']
+        assert stored_columns == expected
+
+    def test__store_columns_tuple(self):
```

Comment:
We need to distinguish 2 test cases here:
- The tuple exists as a column name within the data frame
- The tuple does not exist as a column within the data frame (but each element within the tuple does)